shared/src/types/ethereum_events.rs

```diff
@@ -219,3 +229,138 @@ pub struct TokenWhitelist {
     /// Maximum amount of token allowed on the bridge
     pub cap: Amount,
 }
+
+/// Represents an Ethereum event being seen by some validators
+#[derive(
+    Debug,
+    Clone,
+    Ord,
+    PartialOrd,
+    PartialEq,
+    Eq,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+)]
+pub struct EthMsgUpdate {
+    /// the event being seen
+    pub body: EthereumEvent,
+    /// addresses of the validators who have just seen this event
+    /// we use [`BTreeSet`] even though ordering is not important here, so that
+    /// we can derive [`Hash`] for [`EthMsgUpdate`]
+    pub seen_by: BTreeSet<Address>,
```

Comment:
Further up the stack the `/eth_msgs/<hash>/seen_by` is still a `Vec<Address>` but maybe a `BTreeSet` would be better?